prediction_market_agent/agents/think_thoroughly_agent/pinecone_utils.py

```diff
@@ -0,0 +1,22 @@
+import typing as t
+
+from prediction_market_agent_tooling.markets.omen.data_models import OmenMarket
+
+from prediction_market_agent.agents.think_thoroughly_agent.models import (
+    PineconeMetadata,
+)
+
+
+def create_texts_from_omen_markets(markets: list[OmenMarket]) -> list[str]:
+    return [m.question_title for m in markets]
+
+
+def create_metadatas_from_omen_markets(
+    markets: list[OmenMarket],
+) -> list[dict[str, t.Any]]:
+    return [
+        PineconeMetadata.model_validate(
+            {"question_title": m.question_title, "market_address": m.id}
+        ).dict()
+        for m in markets
+    ]
```

Comment:
I don't understand this one, so it

1. Creates dictionary `{"question_title": m.question_title, "market_address": m.id}`
2. Creaets PineconeMetadata out of it
3. Converts it back to dictionary that should be same as in 1.?
4. Returns `dict[str, t.Any]` even though we know the exact structure?

Instead of this extra function, wdyt about doing this instead of calling  `create_metadatas_from_omen_markets`?

```
metadatas = [PineconeMetadata.from_omen_market(market).model_dump() for market in markets] 
```

(`dict` method is deprecated, `model_dump` should be used instead)

Reply:
Didn't know `dict()` was deprecated, thanks for the heads-up.
Sure, refactored.